app/src/main/java/org/mozilla/fenix/compose/tabstray/TabGridItem.kt

```diff
@@ -0,0 +1,281 @@
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextDirection
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.core.text.BidiFormatter
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.state.state.createTab
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Favicon
+import org.mozilla.fenix.compose.HorizontalFadingEdge
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Tab grid item used to display a tab that supports clicks,
+ * long clicks, multiple selection, and media controls.
+ *
+ * @param tab The given tab to be render as view a grid item.
+ * @param isSelected Indicates if the item should be render as selected.
+ * @param multiSelectionEnabled Indicates if the item should be render with multi selection options,
+ * enabled.
+ * @param multiSelectionSelected Indicates if the item should be render as multi selection selected
+ * option.
+ * @param onCloseClick Callback to handle the click event of the close button.
+ * @param onMediaClick Callback to handle when the media item is clicked.
+ * @param onClick Callback to handle when item is clicked.
+ * @param onLongClick Callback to handle when item is long clicked.
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+@Suppress("MagicNumber", "LongParameterList", "LongMethod")
+fun TabGridItem(
+    tab: TabSessionState,
+    isSelected: Boolean = false,
+    multiSelectionEnabled: Boolean = false,
+    multiSelectionSelected: Boolean = false,
+    onCloseClick: (tab: TabSessionState) -> Unit,
+    onMediaClick: (tab: TabSessionState) -> Unit,
+    onClick: (tab: TabSessionState) -> Unit,
+    onLongClick: (tab: TabSessionState) -> Unit,
+) {
+    val tabBorderModifier = if (isSelected && !multiSelectionEnabled) {
+        Modifier.border(
+            4.dp,
+            FirefoxTheme.colors.borderAccent,
+            RoundedCornerShape(12.dp)
+        )
+    } else {
+        Modifier
+    }
+
+    Box(
+        modifier = Modifier
+            .wrapContentHeight()
+            .wrapContentWidth()
+    ) {
+        Card(
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(202.dp)
+                .padding(4.dp)
+                .then(tabBorderModifier)
+                .padding(4.dp)
+                .combinedClickable(
+                    onLongClick = { onLongClick(tab) },
+                    onClick = { onClick(tab) }
+                ),
+            elevation = 0.dp,
+            shape = RoundedCornerShape(dimensionResource(id = R.dimen.tab_tray_grid_item_border_radius)),
+            border = BorderStroke(1.dp, FirefoxTheme.colors.borderPrimary)
+        ) {
+            Column(
+                modifier = Modifier.background(FirefoxTheme.colors.layer2)
+            ) {
+                Row(
```

Comment:
Could this `Column` and `Row` potentially be merged?

Reply:
Both the column and the row have multiple children, so this nesting is required to obtain the intended result.

Reply:
My bad, I completely missed where the `Divider` and `Thumbnail` were in the hierarchy.